### PR TITLE
Add visually hidden header to top navigation

### PR DESCRIPTION
**Pull request by kendallgassner**
### What

Each navigation section should have a header to allow users to quickly jump between sections with their assistive technology.

Also, I noticed the top navigation section is not wrapped in a semantic `nav` tag so I added a `nav`. I also changed the react components of the drop-down nav buttons to use the `ActionMenu` and `ActionList` components instead of detail/summary components (the semantics for these components are a bit more accessible). It definintely changes the look of this navigation section so let me know what you think/if we need to change the ActionMenu button text color. 

I did not tackle responsive mode.

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "version": "1.0.0",
   "license": "MIT",
   "scripts": {
-    "develop": "gatsby develop -H 0.0.0.0",
+    "clean": "gatsby clean",
+    "develop": "gatsby develop",
```
Comment (kendallgassner) [inline docs/package.json:8]:
```suggestion
    "develop": "gatsby develop -H 0.0.0.0",
```

File: theme/src/components/nav-dropdown.js [hunk 2]
```diff
@@ -1,36 +1,20 @@
...
-            borderColor: 'border.default',
-            borderRadius: '12px'
-          }}
-        >
-          {children}
-        </Box>
-      </Box>
-    </Details>
+    <ActionMenu {...getDetailsProps()}>
+      <ActionMenu.Button variant="invisible">{title}</ActionMenu.Button>
```
Comment (colebemis) [inline theme/src/components/nav-dropdown.js:9]: Looks like the dropdown items don't match non-dropdown items in the nav now: 
CleanShot 2023-06-29 at 14 40 20

Comment (kendallgassner) [reply theme/src/components/nav-dropdown.js:9]: Yes! I mentioned this in the description should I just change the color to match the other nav items? Is that okay?

**Comment by kendallgassner** [comment]
@colebemis I matched font-weights, font-sizes, and colors of the navigation elements.

To be honest I wonder if it would have been better to just change the links to be blue (the default color) to match the invisible button colors -- that would feel less hacky. Let me know what you think.

My least favorite part of my current method is overridding the button's hover color -- I just needed the contrast to pass.

**Review by langermank — APPROVED** [review]
I understand the concern about the overrides, but this nav will be refactored soon for the new docs design. I think this is a fine solution for now!

**Pull request merged**